types/algoliasearch/index.d.ts

```diff
@@ -1842,8 +1843,8 @@ declare namespace algoliasearch {
     cursor?: string;
   }
 
-  interface MultiResponse {
-    results: Response[];
+  interface MultiResponse<T=any> {
```

Comment:
The signature of the function is not helpful when we target different indices. The type of the record is different in most cases. With this signature the response only contain records of the same type.

Reply:
I don't think this is a particular problem since you could simply define a union type as `T`, like in this example:

![image](https://user-images.githubusercontent.com/439899/52950353-2b30ea80-337f-11e9-911b-cf22afda9a36.png)

Reply:
@keichinger is right. Either way, if you just have too many types, you can also leave blank like before:

`client.search(...)`

The `T` type is optional.

Reply:
Yep, but you have to use a type guard to safely access your hit from the response. What I meant with my comment is: with this signature the response is not tightly coupled to a specific type which is the case with the single query signature.

Reply:
@samouss All good?

Reply:
Yes it’s better than nothing.